### ISSUE TITLE
api to manage xpubs

### ISSUE THREAD

**Issue by benma** [chris-belcher/electrum-personal-server#71]
It would be nice if an electrum-personal-server client could add and remove xpubs through some sort of API, instead of messing with the config file, and without restarting the server.

This would be nice so electrum clients could automatically start tracking an xpub, without the user having to manually dealing with this.

Possibly related: #63

**Comment by chris-belcher** [comment]
I was talking to Nicolas Dorier about BTCPay which could use this feature to make it easier to use EPS with BTCPay. I think they only need a way to add new empty master public keys.

This is maybe done most cleanly by having EPS periodically reload the config file and check whether any new master public keys have been added.

**Comment by benma** [comment]
That would be a good start. How would the proposed workflow be from a client to add an xpub? If EPS doesn't expose an API call, another sort of service would need to run alongside EPS, right?

**Comment by chris-belcher** [comment]
I don't think requiring another service to be running makes the most sense. It might be better to have EPS itself expose an RPC socket which supports those API calls.

Ideally we'd keep code complexity down and usability up. I think some more on the best way to do this.

**Comment by benma** [comment]
Agreed. Maybe you could extend the electrum protocol with new api endpoints (prefixed with `eps` or similar to keep namespaces separate from the other electrum endpoints), then the same connection can be used. A separate connection just for EPS is also ok, but it would be good to stick to the same rpc protocol and not something different.

**Comment by chris-belcher** [comment]
> How would the proposed workflow be from a client to add an xpub?

I realized I never answered this, sorry.

The workflow would be that the user or software like BTCPay server edits the config file to add a new xpub key. Soon EPS will notice that edit and then start watching the bitcoin addresses corresponding to the xpub and start monitoring for new transactions if any arrive.
